crates/distribution-types/src/file.rs

```diff
@@ -56,10 +56,9 @@ impl File {
                 .map_err(|err| FileConversionError::RequiresPython(err.line().clone(), err))?,
             size: file.size,
             upload_time_utc_ms: file.upload_time.map(|dt| dt.timestamp_millis()),
-            url: if split_scheme(&file.url).is_some() {
-                FileLocation::AbsoluteUrl(file.url)
-            } else {
-                FileLocation::RelativeUrl(base.to_string(), file.url)
+            url: match Url::parse(&file.url) {
```

Comment:
This looks like added parsing. Is it strictly necessary?

Reply:
I'm not 100% sure. I don't think `split_scheme` guarantees that the URLs is necessarily valid, so if it wasn't we would end up erroring later. A `UrlString` should always be a valid `Url`, so I think we need to do this. We might just error a bit earlier than we would before.